include/boost/json/detail/impl/string_impl.ipp

```diff
@@ -12,6 +13,7 @@
 
 #include <boost/json/error.hpp>
 #include <boost/json/detail/string_impl.hpp>
+#include <boost/json/detail/impl/string_impl.hpp>
```

Comment:
I don't think you should be including this here, it is already included in `<boost/json/detail/string_impl.hpp>`

Reply:
Fixed; `ptr_in_range` was turned into an `inline` non-template function and moved into `detail/string_impl.hpp`